### PR TITLE
fixes ui issues with setup, gameboard, and pieces

### PR DESCRIPTION
**Pull request by fireteam99**
# Description

- Fixes https://github.com/chinese-board-games/luzhanqi-web/issues/74
- Fixes https://github.com/chinese-board-games/luzhanqi-web/issues/71
- Updates background color for game board
- Sets reasonable max-widths for game board and half board
- Disables piece selection when not user's turn
## Type of Change

- [ ] New feature
- [x] Bug fix
- [ ] Breaking change
- [ ] Documentation update

## Checklist

- [x] Unit/integration tests
- [x] Documentation

## Screenshots
Game board:
image

Half board:
image

Please attach any design screenshots if UI update.

### CODE REVIEW

File: src/views/Game.js [hunk 1]
```diff
@@ -88,7 +88,7 @@ const Game = () => {
           alignItems: 'center',
           backgroundColor: '#d0edf5'
         }}>
-        <div style={{ width: '35em' }}>
+        <div style={{ width: '100%' }}>
```
Comment (fireteam99) [inline src/views/Game.js:91]: Delegate with constraint responsibility to children components

File: src/utils/custom.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    return false;
+  }
+
+  // landmines can only go into last two rows
+  if (piece.name === 'landmine' && row != 5 && row != 4) {
+    return false;
+  }
+
+  return true;
+};
```
Comment (fireteam99) [inline src/utils/custom.js:25]: This is a frontend exclusive util function. Separating it out will make it easier to migrate back to a shared util package between FE/BE in the future.

**Comment by iankhou** [comment]
Board still doesn't support mobile, but we can scale it down in a different PR.

**Review by iankhou — APPROVED** [review]
Looks good, we need to resize the piece or the spots because they are too similar, and the piece has border radius while the spot does not. [Issue Link #84](https://github.com/chinese-board-games/luzhanqi-web/issues/84)

**Pull request merged**